kiwi/path.py

```diff
@@ -60,6 +67,38 @@ def sort_by_hierarchy(cls, path_list):
                 ordered_paths.append(path)
         return ordered_paths
 
+    @classmethod
+    def access(cls, path, mode, **kwargs):
+        """
+        Check whether path can be accessed with the given mode.
+
+        :param str path: The path that should be checked for
+            access.
+
+        :param int mode: Which access mode should be checked.
+            This value must be a bit-wise or of one or more of the following
+            constants: :py:const:`os.F_OK` (note that this one is zero),
+            :py:const:`os.X_OK`, :py:const:`os.R_OK` and :py:const:`os.W_OK`
+
+        :param kwargs: further keyword arguments are forwarded to
+            :func:`os.access`
+
+        :return: Boolean value whether this access mode is allowed
+        :rtype: bool
+
+        :raises ValueError: if the supplied mode is invalid
+        :raises kiwi.exceptions.KiwiFileNotFound: if the path does not exist or
+            is not accessible by the current user
+        """
+        if mode & ~(os.F_OK | os.X_OK | os.R_OK | os.W_OK) != 0:
+            raise ValueError("Invalid mode 0x{:X}".format(mode))
+        try:
+            os.stat(path)
+        except StatExceptionFileNotFound:
+            raise KiwiFileNotFound("Path {!s} not found".format(path))
```

Comment:
```python
except (OSError, FileNotFoundError) as issue:
    raise KiwiFileNotFound(
        'Error accessing path {0}: {1}'.format(path, issue)
    )
```

Reply:
I opted not to do this, since that will catch `OSError` for Python 3 too, which we might not want.

Reply:
I'm just thinking why we don't want to catch OSError. If os.stat raised with an OSError I think we should catch that in any way. This would also be the reason why I tried to make the error message a bit more generic and use the exception information as part of the error message.

Actually if I think further down is there any reason why we have to specifically watch out for an exception type ? I mean if os.stat throws an exception it should be catched, or is there any raise condition which we want to handle later ? I think if we let pass one here it will end up in a raise at another code path that might be misleading or in worst case ends in a stack trace.

I found one more. You raise a ValueError but that is a python exception and outside of the scope of kiwi. This will surely lead to a stack trace and is treated as an "unknown error", but we want to treat it as a known error. So I suggest the following:

1. add exceptions to kiwi/exceptions.py

```python
class KiwiValueError(KiwiError):
    """
    Exception raised if an invalid value for an operation is used
    """


class KiwiFileAccessError(KiwiError):
    """
    Exception raised if accessing file metadata has failed
    """
```

2. Now in the code

```python
    if mode & ~(os.F_OK | os.X_OK | os.R_OK | os.W_OK) != 0:
￼        raise KiwiValueError('Invalid mode 0x{:X}'.format(mode))
    try:
        os.stat(path)
￼    except Exception as issue:
        raise KiwiFileAccessError(
            'Error accessing path {0}: {1}'.format(path, issue)
        )
```

Thoughts ?

Reply:
1. I've added the raising of `ValueError` as a safety net for the programmer. This is something that a static type system could catch, but since we don't have that, I have opted for an exception instead. This part should only be reachable if a programmer messes up and passes a wrong bitmask to `Path.access`. Imo that should result in a backtrace, as that will make it obvious that that's a bug and not some error that the user inserted (which most `Kiwi*Error` exceptions indicate).

2. There's not really a reason to not catch `OSError`, I'll include the above code.

Reply:
>  Imo that should result in a backtrace, as that will make it obvious that that's a bug

ah ok

Reply:
I've incorporated the changes from 2. into the PR. 

If you don't like the `ValueError` throwing, I can also include that (it's your project, so it's your decision, I was merely explaining my thought process).

Reply:
That's all fine, thanks